### ISSUE TITLE
New Win11 update (KB5034123) will mess with Win7 Command bar

### ISSUE THREAD

**Issue by NoNameNeeded5** [valinet/ExplorerPatcher#2676]
If you choose to use Win7 Command bar in explorer window, the newest security update (KB5034123) will mess with it.

What happens is that the menu bar will be missing.
You can click on organize and select the menu bar again and it will appear again.
However, as soon as you close the window and reopen an explorer window, it will be gone again.

**Comment by retiredfields** [comment]
Yes, the option to have the "Windows Command Bar (Classic address bar)" no longer works. Address bar is always at the top.

**Comment by Amrsatrio** [comment]
Yes, unfortunately the fix for it doesn't last long as it will make Explorer crash. The feature flag for the file explorer tabs is now no more :(

I do not know if I can fix this though.

**Comment by retiredfields** [comment]
Thanks for the quick reply. I suspected this was the situation.

**Comment by NoNameNeeded5** [comment]
I don't know what you mean by "now we're back to the old behavior before the fix" but what I mean is that the menu bar "File, Edit etc." will be hidden as soon as you install the above mentioned update.
It can be made to show again but the next time you open an explorer window it will be hidden again.

I was actually hoping that this could be fixed.
It's not that it's not working at all, it's just hidden by default.

**Comment by Amrsatrio** [comment]
Maybe you're unaware of this issue: https://github.com/valinet/ExplorerPatcher/issues/2243

**Comment by NoNameNeeded5** [comment]
Oh, I see.

That doesn't sound too good.

But you guys have achieved so much, I'm still hoping that you'll be able to fix this issue too.

**Comment by pyrates999** [comment]
Try using oldNewExplorer to patch file explorer instead and see if it achieves what you want.

**Comment by NoNameNeeded5** [comment]
I've already tried that but no, it didn't help.

**Comment by pyrates999** [comment]
ah ok

**Comment by Anixx** [comment]
Confirming menu bar disappearance. It is not always hidden (it is quite random), so there should be a way to make it always visible.

**Comment by Anixx** [comment]
It looks like the menubar 100% can be restored by changing the current color scheme, specifically, windowmetrics, regardless of what the change is.

Could it be that the menu fail to appear and that the background is initially black because the window fails to get the color scheme data? What if one manually forces the window to update the window metric data after creation?

I tried this by sending messages but for some reason it crashes Explorer for me, so I am doing it the wrong way apparently. On the other hand, the color scheme configurators easily make all opened folder windows to have a menu after sending the theme update message.

Maybe we can try to hook the folder window creation and force it to refresh window metrics and colors?

**Comment by Anixx** [comment]
I have checked it, and my conclusion:

1. This does not depend on ExplorerPatcher running or not.
2. It does not depend on the Win7 Command Bar being activated.

Without any EP and with default Win11 Command bar the menubar appears on any Window Metrics change or refresh, for instance, with simple deskn.cpl, the legacy color scheme CP applet.

**Comment by Amrsatrio** [comment]
Fixed menubar when 7 command bar is used on builds with explorer tabs enabled. Coming in 69.6.

**Comment by Amrsatrio** [comment]
Fixed in 69.6.

**Comment by PandaFiredoge** [comment]
26200.8037，menubar is completely removed

**Comment by Anixx** [comment]
Why are they removing the core things? It seems, removing the start menu did not teach them.

**Comment by pyrates999** [comment]
My guess is that Microsoft internally has listed features that are legacy, but can't be removed due to it being a dependency of another legacy feature.  So they're first removing the legacy features that no other features depends on.  And if that core feature no longer is listed as a dependency to another feature, then it can safely be removed.
